src/qt3utils/datagenerators/daqsamplers.py

```diff
@@ -287,17 +280,97 @@ def _burn_and_log_exception(self, f):
     def stop(self):
         if self.running:
             while self.read_lock:
-                time.sleep(0.1)  # wait for current read to complete
+                time.sleep(0.1) # wait for current read to complete
 
             if self.nidaq_config.clock_task:
                 self._burn_and_log_exception(self.nidaq_config.clock_task.stop)
-                self._burn_and_log_exception(
-                    self.nidaq_config.clock_task.close)  # close the task to free resource on NIDAQ
-            # self._burn_and_log_exception(self.nidaq_config.counter_task.stop) # will need to stop task if we move
-            # to continuous buffered acquisition
+                self._burn_and_log_exception(self.nidaq_config.clock_task.close) # close the task to free resource on NIDAQ
+            # self._burn_and_log_exception(self.nidaq_config.counter_task.stop) # will need to stop task if we move to continuous buffered acquisition
             self._burn_and_log_exception(self.nidaq_config.counter_task.close)
 
         self.running = False
 
     def close(self):
         self.stop()
+
+
+
+class NiDaqTimedDigitalInputRateCounter(NiDaqDigitalInputRateCounter):
+
+    '''
+    This class is an alternative version of its parent class
+    `NiDaqDigitalInputRateCounter` which replaces the number of samples
+    per batch with the time per sample (in seconds) as the fundamental 
+    measure for the sampling time.
+    
+    This enables users to specify the measurement time and configure it
+    at the lowest possible level.
```

Comment:
Bless up

Reply:
FYI, implementation of this is not complete but I will address it in issue #4 most likely.